conans/client/tools/win.py

```diff
@@ -67,25 +68,91 @@ def vs_installation_path(version):
 
     if version not in vs_installation_path._cached:
         vs_path = None
-        program_files = os.environ.get("ProgramFiles(x86)", os.environ.get("ProgramFiles"))
-        if program_files:
-            vswhere_path = os.path.join(program_files, "Microsoft Visual Studio", "Installer",
-                                        "vswhere.exe")
-            if os.path.isfile(vswhere_path):
-                version_range = "[%d.0, %d.0)" % (int(version), int(version) + 1)
-                try:
-                    output = subprocess.check_output([vswhere_path, "-version", version_range,
-                                                      "-legacy", "-property", "installationPath"])
-                    vs_path = output.decode().strip()
-                    _global_output.info("vswhere detected VS %s in %s" % (version, vs_path))
-                except (ValueError, subprocess.CalledProcessError, UnicodeDecodeError) as e:
-                    _global_output.error("vswhere error: %s" % str(e))
+        output_legacy = json.loads(vswhere(legacy_=True, format_="json"))
+        output_products = json.loads(vswhere(products_=["*"], format_="json"))
+        output = output_legacy + output_products
+
+        for installation in output:
+            if installation["installationVersion"].startswith(("%d." % int(version))):
+                vs_path = installation["installationPath"]
 
         # Remember to cache result
         vs_installation_path._cached[version] = vs_path
 
     return vs_installation_path._cached[version]
 
+def vswhere(all_=False, prerelease_=False, products_=list(), requires_=list(), version_="",
```

Comment:
Do not use a list object as default parameter, it is a bug. Use None, then inside the method initialize it

Reply:
Why all those underscores for argument names?

Reply:
Search for "single_trailing_underscore_" in [PEP 8](https://www.python.org/dev/peps/pep-0008/#descriptive-naming-styles)

Reply:
But that is only ``all``, the rest are not python words, are they?

Reply:
all, format and property